### PR TITLE
FrontEnd unitchecking

### PR DESCRIPTION
**Pull request by arun3688**

### CODE REVIEW

File: Compiler/NFFrontEnd/NFUnitCheck.mo [hunk 1]
```diff
@@ -0,0 +1,1394 @@
...
+  NFHashTableCrToUnit.HashTable HtCr2U1, HtCr2U2;
+  NFHashTableStringToUnit.HashTable HtS2U;
+  NFHashTableUnitToString.HashTable HtU2S;
+  DAE.FunctionTree func;
+  list<DAE.Function> functionlist;
+  list<Functionargs> args;
+  list<String> inunits,outunits;
+algorithm
+    try
+			  (elts1,elts2):=DAEUtil.splitDAEIntoVarsAndEquations(indae);
```
Comment (lochel) [inline Compiler/NFFrontEnd/NFUnitCheck.mo]: Why are there tabs mixed into the indentation?

**Comment by lochel** [comment]
Thanks for moving the implementation into the frontend.
What is the status of the front end unit checking? Are there any known issues?

**Comment by arun3688** [comment]
As far i know all the test cases which worked for the backend unitchecking, worked with frontend unitchecking also,

**Comment by lochel** [comment]
@arun3688 wrote:
> As far i know all the test cases which worked for the backend unitchecking, worked with frontend unitchecking also,

This is good news. I will test it soon.

**Comment by arun3688** [comment]
The test cases are in the following path, testsuite/simulation/modelica/NFunitcheck

File: Compiler/Util/Flags.mo [hunk 2]
```diff
@@ -509,6 +509,9 @@ constant DebugFlag PARTITION_INITIALIZATION = DEBUG_FLAG(167, "partitionInitiali
   Util.gettext("This flag controls if partitioning is applied to the initialization system."));
 constant DebugFlag EVAL_PARAM_DUMP = DEBUG_FLAG(168, "evalParameterDump", false,
   Util.gettext("Dumps information for evaluating parameters."));
+constant DebugFlag NF_UNITCHECK = DEBUG_FLAG(169, "frontEndUnitCheck", false,
+  Util.gettext("Checks the consistency of units in equation."));
+
```
Comment (lochel) [inline Compiler/Util/Flags.mo]: There is also the old flag `unitChecking` which is used to activate the old front end implementation. If that’s not used anymore, we could remove that implementation and reuse the old config flag.

**Comment by OpenModelica-Hudson** [comment]
The test suite is unstable according to [OpenModelica_TEST_PULL_REQUEST 2016-12-02_11-17-59](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/3491/).

**Comment by OpenModelica-Hudson** [comment]
The tests run correctly according to [OpenModelica_TEST_PULL_REQUEST 2016-12-02_12-32-02](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/3493/console).

**Comment by casella** [comment]
@arun3688 would you mind updating the status of development on [#4107](https://trac.openmodelica.org/OpenModelica/ticket/4107)?

**Pull request closed without merging**